tests/gbs/test_data.py

```diff
@@ -150,3 +160,52 @@ def test_adj_dim(self, dataset):
     def test_adj_valid(self, dataset):
         """Test if adjacency matrix of dataset is symmetric."""
         assert np.allclose(dataset.adj, dataset.adj.T)
+
+
+@pytest.mark.parametrize("datasets", MOLECULE_DATASETS_LIST)
+class TestMoleculeDatasets:
+    """Tests for the ``MoleculeDataset`` class"""
+
+    @pytest.fixture
+    def dataset(self, datasets):
+        """Fixture for loading each of the datasets in ``MOLECULE_DATASETS_LIST``"""
+        yield datasets()
+
+    def test_w_dims(self, dataset):
+        """Test if w has correct shape"""
+        w, _, _, _ = dataset.w, dataset.wp, dataset.Ud, dataset.delta
+        assert w.shape == (dataset.modes // 2,)
+
+    def test_wp_dims(self, dataset):
+        """Test if wp has correct shape"""
+        _, wp, _, _ = dataset.w, dataset.wp, dataset.Ud, dataset.delta
+        assert wp.shape == (dataset.modes // 2,)
+
+    def test_Ud_dims(self, dataset):
+        """Test if Ud has correct shape"""
+        _, _, Ud, _ = dataset.w, dataset.wp, dataset.Ud, dataset.delta
+        assert Ud.shape == (dataset.modes // 2, dataset.modes // 2)
+
+    def test_delta_dims(self, dataset):
+        """Test if delta has correct shape"""
+        _, _, _, delta = dataset.w, dataset.wp, dataset.Ud, dataset.delta
+        assert delta.shape == (dataset.modes // 2,)
+
+    def test_uniform_dims(self, dataset):
+        """Test if overall dimension is correct"""
+        w, wp, Ud, delta = dataset.w, dataset.wp, dataset.Ud, dataset.delta
+        dim = w.shape[0]
+
+        assert wp.shape[0] == dim and Ud.shape[0] == dim and delta.shape[0] == dim

```

Comment:
```suggestion
        assert wp.shape[0] == dim and Ud.shape == (dim, dim) and delta.shape[0] == dim
```
Otherwise we only check that the number of columns is correct